xarray/backends/zarr.py

```diff
@@ -259,13 +260,13 @@ def encode_zarr_variable(var, needs_copy=True, name=None):
         raise NotImplementedError("Variable `%s` is an object. Zarr "
                                   "store can't yet encode objects." % name)
 
-    var = conventions.maybe_encode_datetime(var, name=name)
-    var = conventions.maybe_encode_timedelta(var, name=name)
-    var, needs_copy = conventions.maybe_encode_offset_and_scale(var,
-                                                                needs_copy,
-                                                                name=name)
-    var, needs_copy = conventions.maybe_encode_fill_value(var, needs_copy,
-                                                          name=name)
+    for coder in [coding.times.CFDatetimeCoder(),
+                  coding.times.CFTimedeltaCoder(),
+                  coding.variables.CFScaleOffsetCoder(),
+                  coding.variables.CFMaskCoder(),
+                  coding.variables.UnsignedCoder()]:
+        var = coder.encode(var, name=name)
+
```

Comment:
@shoyer - what do you think about adding encode/decode methods to the `AbstractWritableDataStore`? It seems each backend handles these slightly differently but this step happens for all backends.

Reply:
Yes, some of these need to get associated with the backend classes in some way. I was waiting to do that until we finish porting all of the stuff in `conventions` into `coding`.